lib/cmds/organization_cmds/utils/taxonomy.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ConceptProps,
+  ConceptSchemeProps,
+  CreateConceptProps,
+  CreateConceptSchemeProps
+} from 'contentful-management'
+import { Concept } from './concept'
+import { ConceptScheme } from './concept-scheme'
+
+export class Taxonomy {
+  private concepts: {
+    existing: Array<Concept>
+    upsert: Array<Concept>
+  }
+
+  private conceptSchemes: {
+    existing: Array<ConceptScheme>
+    upsert: Array<ConceptScheme>
+  }
+
+  constructor() {
+    this.concepts = {
+      existing: [],
+      upsert: []
+    }
+
+    this.conceptSchemes = {
+      existing: [],
+      upsert: []
+    }
+  }
+
+  toJson() {
+    return {
+      concepts: this.concepts.upsert.map(concept => concept.toJson()),
+      conceptSchemes: this.conceptSchemes.upsert.map(conceptScheme =>
+        conceptScheme.toJson()
+      )
+    }
+  }
+
+  setExistingConcepts(concepts: Array<ConceptProps>) {
+    this.concepts.existing = concepts.map(
+      concept => new Concept(concept.sys.id, concept)
+    )
+  }
+
+  setExistingConceptSchemes(conceptSchemes: Array<ConceptSchemeProps>) {
+    this.conceptSchemes.existing = conceptSchemes.map(
+      conceptScheme => new ConceptScheme(conceptScheme.sys.id, conceptScheme)
+    )
+  }
+
+  addConcept(
+    id: string,
+    init: Partial<Omit<CreateConceptProps, 'id'>> & {
+      prefLabel: CreateConceptProps['prefLabel']
+    }
+  ) {
+    if (this.concepts.existing.find(concept => concept.toJson().id === id)) {
+      return null
+    }
+
+    if (this.concepts.upsert.find(concept => concept.toJson().id === id)) {
+      return null
+    }
+
+    const concept = new Concept(id, init)
+
+    this.concepts.upsert.push(concept)
+
+    return concept
+  }
+
+  getConcept(id: string) {
+    const existingConcept = this.concepts.existing.find(
+      concept => concept.toJson().id === id
+    )
+    const addedConcept = this.concepts.upsert.find(
+      concept => concept.toJson().id === id
+    )
+
+    if (!existingConcept || !addedConcept) {
+      return null
+    }
```

Comment:
curious about this bit of logic -  if either `existingConcept` or `addedConcept` is falsy, the method returns `null`, so could this unintentionally prevent the method from returning a concept even though it exists in one of the arrays?

wondering if it should instead be 
```suggestion
    if (!existingConcept && !addedConcept) {
      return null
    }
```
but there might be a desired behavior here I'm missing

Reply:
oh, you are totally right